lib/iris/fileformats/cf.py

```diff
@@ -692,7 +699,7 @@ def cf_label_data(self, cf_data_var):
                 else:
                     label_index = index + (slice(None, None),)
 
-                data[index] = ''.join(label_data[label_index]).strip()
+                data[index] = b''.join(label_data[label_index]).strip().decode()
```

Comment:
I think this would be better with an explicit `.decode('utf-8')`.

The [netCDF docs](https://www.unidata.ucar.edu/software/netcdf/docs/cdl_data_types.html) mention strings being encoded as UTF-8. On Python 3 the default encoding for the `bytes.decode` method is UTF-8, but in Python 2 the default encoding for `str.decode` is ASCII.

@bjlittle - I think you did a bunch of the Unicode handling in Python 2 ... have you any guidance here?